x/mongo/driver/operation.go

```diff
@@ -446,7 +446,19 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 			if e := err.(Error); retryable && op.Type == Write && e.UnsupportedStorageEngine() {
 				return ErrUnsupportedStorageEngine
 			}
-			if retryable && tt.Retryable() && retries != 0 {
+
+			connDesc := conn.Description()
+			retryableErr := tt.Retryable()
```

Comment:
Can this be refactored to not call `tt.Retryable` in the case of writes because it'll be overwritten by the call to `tt.RetryableWrite` directly below? Maybe a switch statement or if-else block.